src/publisher.js

```diff
@@ -130,21 +155,59 @@ Publisher.prototype.publish = function () {
 
 					http(config, function (error, response) {
 						if (!error && (response.statusCode >= 200 && response.statusCode < 300)) {
-							deferred.resolve();
+							publishDeferred.resolve(data);
 						} else {
 							if (error != null) {
-								deferred.reject(error);
+								publishDeferred.reject(error);
 							} else {
-								deferred.reject(new Error('Unable to publish Pact to Broker: ' + response.statusCode));
+								publishDeferred.reject(new Error('Unable to publish Pact to Broker: ' + response.statusCode));
```

Comment:
From what I can tell it's to be able to return a more useful error message. Would you prefer something like:

```js
if (!error && (response.statusCode >= 200 && response.statusCode < 300)) {
  publishDeferred.resolve(data);
else if (error != null) {
  publishDeferred.reject(error);
} else {
  publishDeferred.reject(new Error('Unable to publish Pact to Broker: ' + response.statusCode));
}
```